test/new-e2e/pkg/utils/e2e/client/host.go

```diff
@@ -386,6 +394,60 @@ func (h *Host) getSFTPClient() *sftp.Client {
 	return sftpClient
 }
 
+// HTTPTransport returns an http.RoundTripper which dials the remote host.
+// This transport can only reach the host.
+func (h *Host) HTTPTransport() http.RoundTripper {
+	return h.httpTransport
+}
+
+// HTTPClient returns an *http.Client which dials the remote host.
+// This client can only reach the host.
+func (h *Host) HTTPClient() *http.Client {
+	return &http.Client{
+		Transport: h.httpTransport,
+	}
+}
+
+func (h *Host) newHTTPTransport() *http.Transport {
+	return &http.Transport{
+		DialContext: func(ctx context.Context, _, addr string) (net.Conn, error) {
+			hostname, port, err := net.SplitHostPort(addr)
+			if err != nil {
+				return nil, err
+			}
+
+			// best effort to detect logic errors around the hostname
+			// if the hostname provided to dial is not one of those, return an error as
+			// it's likely an incorrect use of this transport
+			validHostnames := map[string]struct{}{
+				"":                             {},
+				"localhost":                    {},
+				"127.0.0.1":                    {},
+				h.client.RemoteAddr().String(): {},
+			}
+
+			if _, ok := validHostnames[hostname]; !ok {
+				return nil, errors.New("request hostname does not match host address")
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
```suggestion
				return nil, fmt.Errorf("request hostname %s does not match any valid host name", hostname)
```